demos/linear_mixed_system/linear_mixed_system.py.rst

```diff
@@ -0,0 +1,328 @@
+
+After numerous manipulations (described in detail in :cite:`Salwa:2017`) and evaluation of individual variations, the time-discrete equations, with symplectic Euler scheme, that we would like to implement in Firedrake, are:
+
+.. math::
+
+    \begin{align}
+    & \int v \phi^{n+1} \, {\mathrm d} S_f = \int v (\phi^n - \Delta t \eta^n) \, {\mathrm d} S_f \\
+    & \int \rho_0 {\bf v} \cdot {\bf U}^{n+1} \, {\mathrm d} V_S \underline{ + \int {\bf n} \cdot {\bf v} \, \phi^{n+1} \, {\mathrm d} S_s} = \rho_0 \int {\bf v} \cdot {\bf U}^n \, {\mathrm d} V_S \nonumber\\ 
+    & \hspace{5cm}  - \Delta t \int \left( \lambda \nabla \cdot {\bf v} \nabla \cdot {\bf X}^n + \mu \frac{\partial X^n_j}{\partial x_i}  ( \frac{\partial v_i}{\partial x_j}  + \frac{\partial v_j}{\partial x_i} ) \right) \, {\mathrm d} V_S \underline{ + \int {\bf n} \cdot {\bf v} \, \phi^n \, {\mathrm d} S_s }
+    \\
+    & \int \nabla v \cdot \nabla \phi^{n+1} \, {\mathrm d} V_F \underline{ - \int v {\bf n} \cdot {\bf U}^{n+1} \, {\mathrm d} S_s }= 0 \\ %\hspace{1cm} (+ \text{Dirichlet BC at } \partial \Omega_f)\\
+    & \int v \eta^{n+1} \, {\mathrm d} S_f = \int v \eta^n \, {\mathrm d} S_f + \Delta  t \int \nabla v \cdot \nabla \phi^{n+1} \, {\mathrm d} V_F   \underline{ - \Delta t \int v {\bf n} \cdot {\bf U}^{n+1}\, {\mathrm d} S_s }\\
+    & \int {\bf v} \cdot {\bf X}^{n+1} \, {\mathrm d} V_S = \int {\bf v} \cdot ( {\bf X}^n + \Delta t {\bf U}^{n+1} ) \, {\mathrm d} V_S \, .
+    \end{align}
+
+The underlined terms are the coupling terms. Note that the first equation for :math:`\phi` at the free surface is solved on the free surface only, the last equation for :math:`{\bf X}` in the structure domain, while others in both domains. Moreover, the second and third equations for :math:`\phi` and :math:`{\bf U}` need to be solved simultaneously. Geometry of the system with initial condition is shown below.
+
+.. figure:: geometry.png
+   :align: center
+   :alt: Geometry and initial condition.
+
+   Geometry and initial condition in the system. Fluid (blue) with deflected free surface and the structure (red).
+
+Now we present the code used to solve the system of equations above. We start with appropriate imports::
+
+    from firedrake import *
+    import math
+    import numpy as np
+
+Then, we set parameters of the simulation::
+
+    # parameters in SI units
+    t_end = 5. # time of simulation [s]
+    dt = 0.005 # time step [s]
+    g = 9.8 # gravitational acceleration
+    # water
+    Lx = 20. # length of the tank [m] in x-direction; needed for computing initial condition
+    Lz = 10. # height of the tank [m]; needed for computing initial condition
+    rho = 1000. # fluid density in kg/m^2 in 2D [water]
+    # solid parameters
+    rho_B = 7700. # structure density in kg/m^2 in 2D [steel]
+    lam = 1e7 # 9.695e10 for steel, # N/m in 2D - first Lame constant
+    mu = 1e7 # 7.617e10 for steel, # N/m in 2D - second Lame constant
```

Comment:
This reads slightly strangely. You are using values of 1e7 (which I assume is correct) but then you give an explanation for a different value.